Sources/PackageGraph/PackageGraph+Loading.swift

```diff
@@ -383,25 +383,66 @@ private func createResolvedPackages(
     }
 
     // Find duplicate products in the package graph.
-    let duplicateProducts = packageBuilders
-        .flatMap({ $0.products })
-        .map({ $0.product })
-        .spm_findDuplicateElements(by: \.name)
-        .map({ $0[0].name })
-
-    // Emit diagnostics for duplicate products.
-    for productName in duplicateProducts {
-        let packages = packageBuilders
-            .filter({ $0.products.contains(where: { $0.product.name == productName }) })
-            .map{ $0.package.identity.description }
-            .sorted()
-
-        observabilityScope.emit(PackageGraphError.duplicateProduct(product: productName, packages: packages))
-    }
-
-    // Remove the duplicate products from the builders.
-    for packageBuilder in packageBuilders {
-        packageBuilder.products = packageBuilder.products.filter({ !duplicateProducts.contains($0.product.name) })
+    let productList = packageBuilders.flatMap({ $0.products }).map({ $0.product })
+
+    if moduleAliasingUsed {
```

Comment:
@neonichu could you please confirm this works as you expected

Reply:
per discussion, it seems safer to limit this change to automatic library products only.